data/transactions/logic/eval.go

```diff
@@ -4848,7 +4861,7 @@ func (cx *EvalContext) availableRound(r uint64) (basics.Round, error) {
 	if firstAvail > cx.txn.Txn.LastValid || firstAvail == 0 { // early in chain's life
 		firstAvail = 1
 	}
-	current := cx.Ledger.Round()
+	current := cx.SigLedger.Round()
 	round := basics.Round(r)
 	if round < firstAvail || round >= current {
```

Comment:
this is interesting, Round() is either current, or latest, depending on a context, and the condition `round >= current` is technically correct for both contexts but looks as a maintainability problem.

Reply:
I've changed it so that regardless of how the SigLedger is supplied (either as a copy of Ledger, or on its own for LogicSig evaluation) Round() is Latest()+1. I think that is fairly good, although I remain worried about transactions added to a long queue that will be dropped asking for a header that doesn't exist yet.